runtime/compiler/runtime/JITServerAOTSerializationRecords.hpp

```diff
@@ -286,6 +288,36 @@ struct AOTHeaderSerializationRecord : public AOTSerializationRecord
    };
 
 
+struct ThunkSerializationRecord : public AOTSerializationRecord
+{
+public:
+   uint32_t signatureSize() const { return _signatureSize; }
+   uint32_t thunkCodeSize() const { return _thunkCodeSize; }
+   const uint8_t *signature() const { return _varSizedData; }
+   const uint8_t *thunkCode() const { return _varSizedData + _signatureSize; }
+
+private:
+   friend class AOTCacheRecord;
+   friend class AOTCacheThunkRecord;
+
+   ThunkSerializationRecord(uintptr_t id, const uint8_t *signature, uint32_t signatureSize, const uint8_t *thunkCode,  uint32_t thunkCodeSize);
+   ThunkSerializationRecord();
+
+   static size_t size(uint32_t signatureSize, uint32_t thunkCodeSize)
+      {
+      return sizeof(ThunkSerializationRecord) + OMR::alignNoCheck(signatureSize + thunkCodeSize, sizeof(uint32_t));
```

Comment:
The size should still be rounded up to `sizeof(size_t)` - that's the alignment of this struct (including the header). Otherwise these structs will be misaligned in the cache snapshot file. While this currently wouldn't break anything, this could become an issue, e.g. if we ever decide to map the snapshot file into memory instead of doing a copying read.

Reply:
An overzealous `size_t` -> `uint32_t` replacement.